File: Leetcode/typescript/sol1404.ts

// https://leetcode.com/problems/number-of-steps-to-reduce-a-number-in-binary-representation-to-one/?envType=daily-question&envId=2026-02-26

const binToDecimal = (s) =>{
    let decimal = 0n;
    for(let i = 0; i<s.length;i++){
        decimal = decimal * 2n + BigInt(s[i]);
    }
    return decimal;
}

function numSteps(s: string): number {
    let decimal:bigint = binToDecimal(s);
    let count  = 0;

    while (decimal > 1n){
        count++;
        if(decimal%2n == 0n){
            decimal = decimal/2n;
        }else{
            decimal++;
        }
    }
    return count;
};

File: Leetcode/python/sol1404.py

# https://leetcode.com/problems/number-of-steps-to-reduce-a-number-in-binary-representation-to-one/description/

class Solution:
    def binToDecimal(self, s) -> int:
        res = 0
        exp = 0
        for i in s[::-1]:
            # res = res * 2 + int(i)
            res += int(i)*2**exp
            exp+=1
        return res

    def numSteps(self, s: str) -> int:
        # decimal = self.binToDecimal(s)
        decimal = int(s, 2)
        count = 0
        while decimal > 1:
            count += 1
            if decimal % 2 == 0:
                decimal //= 2
            else:
                decimal +=1
        return count